src/components/Common/SearchByMultipleFields.tsx

```diff
@@ -0,0 +1,261 @@
+import React, {
+  useState,
+  useCallback,
+  useRef,
+  useEffect,
+  useMemo,
+} from "react";
+import { useTranslation } from "react-i18next";
+import { Input } from "@/components/ui/input";
+import { Button } from "@/components/ui/button";
+import { cn } from "@/lib/utils";
+import CareIcon from "@/CAREUI/icons/CareIcon";
+import PhoneNumberFormField from "@/components/Form/FormFields/PhoneNumberFormField";
+import { FieldError } from "@/components/Form/FieldValidators";
+import {
+  Command,
+  CommandGroup,
+  CommandItem,
+  CommandList,
+} from "@/components/ui/command";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+
+interface SearchOption {
+  key: string;
+  label: string;
+  type: "text" | "phone";
+  placeholder: string;
+  value: string;
+  shortcutKey: string;
+  component?: React.ComponentType<HTMLDivElement>;
+}
+
+interface SearchByMultipleFieldsProps {
+  options: SearchOption[];
+  onSearch: (key: string, value: string) => void;
+  initialOption?: SearchOption;
+  className?: string;
+  inputClassName?: string;
+  buttonClassName?: string;
+}
+
+export type EventType = {
```

Comment:
This is a react event right, it should have its type already defined right 🤔

Reply:
1. this is used only in this file; let's not export it.
2. type of target could be defined from one of the inbuilt types.

Reply:
Instead of using a custom EventType, we can easily use the `FieldChangeEvent` 
 from `@/components/Form/FormFields/Utils`

```ts
// For regular inputs
const commonInputProps = {
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => {
    handleSearchChange(e.target.value);
  },
  // ... other props
};

// For phone input
const phoneInputProps = {
  ...commonInputProps,
  onChange: (event: FieldChangeEvent<string>) => {
    handleSearchChange(event.value);
  },
};
```
This approach:
- This adds `phoneInputProps` which will solve the type error. 
- Removes the need for the custom EventType
- Properly types both regular input and phone input events
- Better follows React's event handling patterns